policyengine_api/routes/household_routes.py

```diff
@@ -0,0 +1,199 @@
+from flask import Blueprint, Response, request
+import json
+
+from policyengine_api.constants import COUNTRY_PACKAGE_VERSIONS
+from policyengine_api.services.household_service import HouseholdService
+from policyengine_api.utils import hash_object
+from policyengine_api.utils.payload_validators import (
+    validate_household_payload,
+    validate_country,
+)
+
+
+household_bp = Blueprint("household", __name__)
+household_service = HouseholdService()
+
+
+@household_bp.route("/<household_id>", methods=["GET"])
+@validate_country
+def get_household(country_id: str, household_id: str) -> Response:
```

Comment:
I see where you're coming from, and you're right, the documentation on this usage is pretty thin. I would not be against migrating to the format you've described. I think that no matter what format we choose, we're operating in a less than ideal environment, since literally every route requires a country ID value.

I also agree that there's probably a benefit to flattening the API structure overall in the long-run. However, this introduces a couple of important questions. For example, we have one or two triple-nested endpoints (e.g., `<country_id>/household/<household_id>/over/policy/<policy_id>` or something similar), and we'd have to consider a proper implementation for them. 

Additionally, this form of migration is obviously breaking, and because the API wasn't designed with a version marker in its route, our only way of migrating is inherently breaking - we must make changes to the API, break it temporarily, then update the front-end app.

Finally, there are cases of routes where we definitely need the country ID. Metadata is obviously one, but that's not difficult to implement within a flattened API structure. The harder-to-implement routes are those enqueuing society-wide calculation.

My proposed solution here would be to shift to the format you've recommended (and also update the other blueprints added recently for AI and economy endpoints), then engage in a longer discussion regarding flattening the structure.

Reply:
## TLDR
 agree.


## Less than ideal paths
Yeah. As structured now, it's awkward.

## flattening that example
Yeah I mean you could go a bunch of ways there. 

### The Computation is the Entity
IMO we're modeling a request to compute a value based on inputs and modeling this as a POST to a computations entity actually ends up being a lot more flexible/useful than treating this as a child of a household and a policy.

**Create a computation**
```
POST /computations/
{
  household_id=...
  policy_id=...
}
```
returns
a link to /computations/[computation_id]

**GET the result**
```
GET
/computations/[computation_id]
{
    status="..."
    inputs= {
       household
       policy
       ...
    }
    data={...}
}
```

**Caching**
Post simply redirects to an existing computation instead of creating a new one.

In fact the "computation_id" could just be an amalgam of the input values (I.e. [household_id]&[policy_id]) in which case it's easy to map to a cached value.

**Searching**
```
GET /computations?country=...
GET /computations?policy_id=...
GET /computations?household_id=...
```

Pros:
* Puts the one true place for finding computations of all kinds under /computations 
  * on the implementation side I don't need to implement partial queries under household, population, country, etc.
  * on the client side I can ask global questions with one query in one place.
* Simplifies different ways of asking for the same thing: I can easily support both household ID and direct data as input for household without needing another path, just one POST with flexible input. 
* Can all go into a completely encapsulated blueprint that lives under an unambiguous URL.

Cons:
* Client slightly more complicated in that we have POST then GET poll instead of just poll.

## The household (or the policy, or both) is the entity

you could do ``/households/[id]/computations`` and then follow the same model as above

This is less flexible though because
1. I have to have a saved household with a household_id to run the computation (what if I just want to pass the household data directly?)
2. If I want to search for computations across households I have to support that with another endpoint under a different resource.
3. I probably still want to be able to search for computations under the household so now I have to duplicate some of that global search under the household.

Also.. why is the household _the_ entity instead of the policy? I mean it seems reasonable to want to search for all simulations under a policy right? So now do I implement both or pick one, etc?